src/main.rs

```diff
@@ -55,17 +58,17 @@ fn main() -> Result<(), Box<dyn Error>> {
     }
 }
 
-/// Sanatises arg[0]
+/// Sanitises arg[0]
 /// Ensures not run as root
 /// Initialises logger
-/// Parses cli, returning the subcommand of the result
+/// Parses CLI, returning the subcommand of the result
 fn initial_init() -> Result<SubCommand, Box<dyn Error>> {
     if nix::unistd::Uid::current().is_root() {
         // TODO: this pre-empts automation. something to think about
         return Err("This program should not be run as root!".into());
     };
 
-    // sanatise executable name
+    // sanitise executable name
     let args = std::env::args();
     let mut args = args.peekable();
     let Some(fst) = args.peek() else {
```

Comment:
l. 74: `if !fst.ends_with("nixos-rsbuild")`

`ends_with` doesn't seem to be the right check here to me. Wouldn't hat give a false positive for something like:
`nixos-rsbuild build 'whatever#mysystem' nixos-rsbuild`. Which of course is a user-error, too, but it would emit a wrong error message.

Reply:
I'll have to double check, but I _believe_ that `fst` in this case would be `nixos-rsbuild`. std::env::args()` gives you an iterator over ags similar to how a loop over $@ with shift 1 can iterate over the cli args.

This domain isn't my strength, so edge-cases, bad-assumptions, etc. are bound to be an issue here.

I'll look into it.

Reply:
https://doc.rust-lang.org/stable/std/env/fn.args.html

I think we are fine. I should probably explain why we are doing this: safeguard name collision with `nixos-rebuild` in case someone has the bright idea of doing something like `mv (which nixos-rsbuild) (which nixos-rebuild)`. At least, that was my thinking at the time, if i recall correctly.